alphazero/mcts/mcts.py

```diff
@@ -257,18 +274,24 @@ def backpropagate(leaf_node: MCTSNode):
         """
         if not rollout:
             _, v_leaf = leaf_node.calc_policy_value(
-                rollout=rollout, network=net
+                rollout=rollout, network=net, game=game
             )  # Think about batching/efficiency
         else:
             _, v_leaf = leaf_node.calc_policy_value(rollout=rollout, game=game)
-
+        leaf_player = leaf_node.player
         leaf_node.is_terminal = game.is_game_over(leaf_node.state)
         if not leaf_node.is_terminal:
-            leaf_node.expand(game)
+            leaf_node.expand(
+                game,
+                dirichlet_eps=dirichlet_eps,
+                dirichlet_conc=dirichlet_conc,
+            )
         current = leaf_node
-        while not current.is_root:
+        while current is not None:
             current.n_visit += 1
-            current.total_value += v_leaf
+            current.total_value += (
+                v_leaf if current.player == leaf_player else -v_leaf
```

Comment:
Would the following be faster (avoiding `if` statements ) ?
```python
current.total_value += v_leaf
v_leaf *= -1
```

Reply:
Hm, this works for 2 player games but for more than 2 players I'm not sure it would work since flipping the sign every time would not work.